### PR TITLE
Modify rules S1774,S3358: Add exception for C++11 constexpr functions

### PR DESCRIPTION
**Pull request by tomasz-kaminski-sonarsource**

### CODE REVIEW

File: rules/S1774/cfamily/rule.adoc [hunk 1]
```diff
@@ -1,5 +1,12 @@
 include::../rule.adoc[]
 
+== Exceptions
+
+For {cpp}11 mode only, the issue is not raised for ternary operators used inside `constexpr` functions. 
+In {cpp}11 such functions are limited to single statement, that is `return` so use of ternary operator
```
Comment (frederic-tingaud-sonarsource) [inline rules/S1774/cfamily/rule.adoc:6]:
```suggestion
In {cpp}11 such functions are limited to a single return statement, so the use of a ternary operator
```

File: rules/S3358/cfamily/rule.adoc [hunk 2]
```diff
@@ -22,6 +22,13 @@ int max(int p1, int p2, int p3) {
 }
 ----
 
+== Exceptions
+
+For {cpp}11 mode only, the issue is not raised for ternary operators used inside `constexpr` functions. 
+In {cpp}11 such functions are limited to single statement, that is `return` so use of ternary operator
```
Comment (frederic-tingaud-sonarsource) [inline rules/S3358/cfamily/rule.adoc:28]:
```suggestion
In {cpp}11 such functions are limited to a single return statement, so the use of a ternary operator
```

**Review by frederic-tingaud-sonarsource — COMMENTED** [review]
Proposing a slight rewording.

File: rules/S1774/cfamily/rule.adoc (hunk 1 above)
Comment (tomasz-kaminski-sonarsource) [reply rules/S1774/cfamily/rule.adoc:6]: This does not convey wording. Being limited to `single return statement` can be read as having multiple statements, but only one `return`. But this function can have only one statement, that is also `return`.

Comment (tomasz-kaminski-sonarsource) [reply rules/S1774/cfamily/rule.adoc:6]: Looks good.

**Review by frederic-tingaud-sonarsource — APPROVED** [review]

**Pull request merged**